rebasehelper/specfile.py

```diff
@@ -352,47 +352,78 @@ def sanitize(tag):
+                m = tag_re.match(expanded)
+                if m:
+                    start = line.find(m.group('prefix'))
+                    if start < 0:
+                        # tag is probably defined by a macro, just ignore it
+                        continue
+                    # conditionalized tag
+                    line = line[start:].rstrip('}')  # FIXME: removing trailing braces is not very robust
+                    m = tag_re.match(line)
+                    if m:
+                        span = cast(Tuple[int, int], tuple(x + start for x in m.span('value')))
+                        if [p for p in parsed if p == expanded.rstrip()]:
+                            result[section].append((sanitize(m.group('name')), index, span))
         return result
 
-    def get_raw_tag_value(self, tag: str) -> Optional[str]:
-        if tag not in self.tags:
+    def _get_matching_section(self, subpackage: Optional[str] = None) -> str:
+        default = '%package'
+        if not subpackage:
+            return default
+        for section in self.section_tags:
+            if subpackage in section:
+                return section
+        return default
+
+    def tags(self, subpackage: Optional[str] = None) -> List[Tag]:
+        return self.section_tags[self._get_matching_section(subpackage)]
+
+    def tag(self, tag_name: str, subpackage: Optional[str] = None) -> Optional[Tag]:
+        tags = [t for t in self.tags(subpackage) if t[0] == tag_name]
+        if tags:
+            return tags[0]  # only care about the first non-unique tag
+        return None
+
+    def get_raw_tag_value(self, tag_name: str, subpackage: Optional[str] = None) -> Optional[str]:
+        tag = self.tag(tag_name, subpackage)
+        if not tag:
             return None
-        index, span = self.tags[tag]
-        return self.spec_content.section('%package')[index][slice(*span)]
+        section = self._get_matching_section(subpackage)
+        _, lineno, span = tag
+        return self.spec_content.section(section)[lineno][slice(*span)]
 
-    def set_raw_tag_value(self, tag: str, value: str) -> None:
-        if tag not in self.tags:
+    def set_raw_tag_value(self, tag_name: str, value: str, subpackage: Optional[str] = None) -> None:
+        tag = self.tag(tag_name, subpackage)
+        if not tag:
             return
-        index, span = self.tags[tag]
-        line = self.spec_content.section('%package')[index]
-        self.spec_content.section('%package')[index] = line[:span[0]] + value + line[span[1]:]
+        section = self._get_matching_section(subpackage)
+        tag_name, lineno, span = tag
+        line = self.spec_content.section(section)[lineno]
+        self.spec_content.section(section)[lineno] = line[:span[0]] + value + line[span[1]:]
         # update span
-        self.tags[tag] = (index, (span[0], span[0] + len(value)))
+        self.tags(subpackage)[0] = (tag_name, lineno, (span[0], span[0] + len(value)))
```

Comment:
Why `self.tags(subpackage)[0]`? Unless I am missing something, `self.tags` returns a list of tags (tuples). The tag to be modified can be on any index in the list, can't it?

Reply:
No, because `SpecFile.tag()` returns only the first one. It a bit hackish, but I don't see any other solution than returning also index from `SpecFile.tag()` (that would defeat the simplicity) or reimplementing its functionality here.

Reply:
I understand that `SpecFile.tag()` returns only the first occurrence but this line that modifies the span calls `SpecFile.tags()`. Shouldn't it then rather be `self.tag(tag_name, subpackage)[0] = ...`

Reply:
No, the code is replacing the first item in the list returned from `SpecFile.tags()`. `SpecFile.tag()` returns a tuple, that is, one, immutable, and two, even if it was mutable the change wouldn't propagate back to the list.